buildSrc/src/main/java/quilt/internal/tasks/lint/SpellingChecker.java

```diff
@@ -0,0 +1,190 @@
+            checkJavadoc(mapping.javadoc(), errorReporter);
+        }
+    }
+
+    private static void checkJavadoc(String javadoc, ErrorReporter errorReporter) {
+        if (javadoc != null && !javadoc.isEmpty()) {
+            // in this case we just have to run the world's slowest regex, there's no clever optimisations based on type to be had
+            List<String> splitJavadoc = splitWords(List.of(javadoc), word -> true, "[0-9_\\[\\]\\-\\.\\{\\}\\(\\)@#/\":; ,<>\n=\\*\\|\\+%`§&!]");
+            for (String word : splitJavadoc) {
+                if (!word.isBlank()) {
+                    checkWord(word.toLowerCase(), errorReporter, "javadoc");
+                }
+            }
+        }
+    }
+
+    private static void checkMapping(Entry<?> entry, String name, ErrorReporter errorReporter) {
+        List<String> namesToSplit = new ArrayList<>();
+
+        // a contains check is necessary here because inner classes do not have package data
+        if (entry instanceof ClassEntry && name.contains(String.valueOf(PACKAGE_SEPARATOR))) {
+            // add class name - we don't have to handle underscores
+            namesToSplit.add(name.substring(name.lastIndexOf(PACKAGE_SEPARATOR) + 1));
+            // add package names and handle underscores
+            String[] packageNames = name.substring(0, name.lastIndexOf(PACKAGE_SEPARATOR)).split(String.valueOf(PACKAGE_SEPARATOR));
+            for (String packageName : packageNames) {
+                String[] split = packageName.split("_");
+                namesToSplit.addAll(Set.of(split));
+            }
+        } else {
+            // handle underscores
+            String[] split = name.split("_");
+            namesToSplit.addAll(List.of(split));
+        }
+
+        List<String> splitNames = splitWords(namesToSplit, word -> word.chars().anyMatch(Character::isDigit), "\\d");
+
+        // check
+        for (String word : splitNames) {
+            checkWord(word, errorReporter, "entry name");
+        }
+    }
+
+    /**
+     * Splits provided words by camelCase, then proceeds to split them by the provided regex if the stripping check succeeds
+     * @param words the words to split
+     * @param strippingCheck a check to run before splitting via the regex
+     * @param strippingRegex a regex to split words by
+     * @return a list of split words
+     */
+    private static List<String> splitWords(List<String> words, Predicate<String> strippingCheck, String strippingRegex) {
+        // split by uppercase characters and numbers and preserve
+        List<String> splitNames = new ArrayList<>();
+        for (String nameToSplit : words) {
+            // split by uppercase letters and preserve them in the split strings
+            // also map to all lowercase
+            Set<String> split = Arrays.stream(nameToSplit.split("(?<!(^|[A-Z]))(?=[A-Z])|(?<!^)(?=[A-Z][a-z])")).map(String::toLowerCase).collect(Collectors.toSet());
```

Comment:
This could look behind for any letter and ahead for an uppercase (i.e. `(?<=[a-zA-z])(?=[A-Z])`), instead of a lowercase behind and an uppercase ahead (`(?<!(^|[A-Z]))(?=[A-Z])`) or an uppercase and a lowercase ahead (`((?<!^)(?=[A-Z][a-z])`)

Reply:
I'll experiment, I remember having a lot of trouble picking the right regex here

Reply:
yours just works :p, feeling a little silly for making something so ridiculous

Reply:
Nah, don't worry, happens every time with regex :P